### DISCUSSION TITLE
How to order using `<->`, the PostgreSQL pg_trgm distance operator

### DISCUSSION THREAD

**Discussion by Pomax** [sequelize/sequelize#16052] [Help & Questions] [answered]
How do I make Sequelize use the PostgreSQL `pg_trgm` distance operator, `<->`, in an `order` statement? I'm trying to effect SQL like

```sql
SELECT * FROM users WHERE login LIKE '%value%' ORDER BY login <-> 'value' LIMIT 20;
```

but the only way I can find is to do this using a literal:

```js
import { literal } from "sequelize";
...  
const users = await this.options.models.User.findAll({
  where: {
    login: {
      [Op.like]: `%${value}%`
    },
  },
  order: literal(`login <-> '${value}'`),
  limit: 20
});
```

which is not ideal, as `literal` by definition does not escape values. I don't see any documentation talking about whether `order` can take the same kind of structure that `where` can (the docs only say order can take objects with a `raw` property), so I have no idea if this is possible:

```js
import { literal } from "sequelize";
...  
const users = await this.options.models.User.findAll({
  where: {
    login: {
      [Op.like]: `%${value}%`
    },
  },
  order: {
    login: {
      "<->": value
    }
  }, 
  limit: 20
});
```

How do I do this? (and where in the docs would I find this?)

**Comment by ephys** [comment, marked as answer]
It requires using a literal

In sequelize 6, you can escape values by calling `sequelize.escape`

In [sequelize 7](https://github.com/sequelize/website/pull/330#issuecomment-1455038636), you can use the [`sql` tag](https://6404627a0dfed06840d8efb6--sequelize-site.netlify.app/docs/v7/querying/raw-queries/) for literals which automatically escapes values

**Comment by Pomax** [comment]
Cheers. It might be worth having this mentioned somewhere in the docs too (probably in the ordering section?), since full text search using GIN/GIST in PostgreSQL is relatively common, but not universal.

**Reply by ephys** [reply]
It is also mentioned in the section about the `order` option in the new docs :) https://6404627a0dfed06840d8efb6--sequelize-site.netlify.app/docs/v7/querying/select-in-depth/#ordering
